lib/service/service.go

```diff
@@ -4542,6 +4542,7 @@ func (process *TeleportProcess) initProxyEndpoint(conn *Connector) error {
 	}
 
 	tlscfg := serverTLSConfig.Clone()
```

Comment:
I notice this `serverTLSConfig` is being `Clone`-ed/used in multiple places in `initAuthService`, however its unclear why we prefer that in some places vs getting a fresh config with `connector.ServerIdentity.TLSConfig(cfg.CipherSuites)` in others. 

Given that `Clone` is a "shallow" clone, I suspect its safer to just create a fresh config for each separate step, unless we explicitly want the pointer-field-dependency of shallow clones.